### PR TITLE
Sunny/e2e field visibility

### PR DESCRIPTION
**Pull request by manivoxel51**
## What changes are proposed in this pull request?

Core e2e test for selection and search for field visibility

example run

https://github.com/voxel51/fiftyone/assets/109545780/d597911c-ac40-41d7-83ad-c444495d1e05

## How is this patch tested? If it is not, please explain why.

(Details)

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: e2e-pw/src/oss/poms/field-visibility/field-visibility.ts [hunk 1]
```diff
@@ -56,11 +171,115 @@ export class FieldVisibilityPom {
...
     return this.modalContainer().getByTestId("field-visibility-btn-apply");
   }
+
+  getFieldInfoContainer(path: string) {
+    return this.modalContainer().getByTestId(
+      `schema-selection-info-container-${path}`
+    );
+  }
+
+  getResetBtn() {
```
Comment (benjaminpkane) [inline e2e-pw/src/oss/poms/field-visibility/field-visibility.ts:181]:
```suggestion
  get resetBtn() {
```

File: e2e-pw/src/oss/poms/field-visibility/field-visibility.ts [hunk 2]
```diff
@@ -56,11 +171,115 @@ export class FieldVisibilityPom {
...
+
+  getTab(tabName: TabType) {
+    return this.modalContainer().getByTitle(tabName);
+  }
+
+  async openTab(tabName: TabType) {
+    return await this.getTab(tabName).click();
+  }
+
+  getFilterRuleContainer() {
```
Comment (benjaminpkane) [inline e2e-pw/src/oss/poms/field-visibility/field-visibility.ts:197]:
```suggestion
  get filterRuleContainer() {
```

File: e2e-pw/src/oss/poms/field-visibility/field-visibility.ts [hunk 3]
```diff
@@ -56,11 +171,115 @@ export class FieldVisibilityPom {
...
+    return this.modalContainer().getByTestId("filter-rule-container");
+  }
+
+  getFilterRuleInput() {
```
Comment (benjaminpkane) [inline e2e-pw/src/oss/poms/field-visibility/field-visibility.ts:201]:
```suggestion
  get filterRuleInput() {
```

File: e2e-pw/src/oss/poms/field-visibility/field-visibility.ts [hunk 4]
```diff
@@ -56,11 +171,115 @@ export class FieldVisibilityPom {
...
+  }
 }
 
 class FieldVisibilityAsserter {
   constructor(
     private readonly fv: FieldVisibilityPom,
     private readonly sb: SidebarPom
   ) {}
+
+  async fieldVisibilityIconHasTooltip() {
```
Comment (benjaminpkane) [inline e2e-pw/src/oss/poms/field-visibility/field-visibility.ts:219]:
```suggestion
  async verifyFieldVisibilityIconHasTooltip() {
```

File: e2e-pw/src/oss/poms/field-visibility/field-visibility.ts [hunk 5]
```diff
@@ -24,13 +58,120 @@ export class FieldVisibilityPom {
...
+        break;
+    }
+
+    const checked = status === "checked";
+    const fields: string[] = [];
+
+    for (let i = 0; i < paths.length; i++) {
+      const cc = this.getFieldCheckbox(paths[i]);
+      if (status === "all") {
+        fields.push(cc);
```
Comment (benjaminpkane) [inline e2e-pw/src/oss/poms/field-visibility/field-visibility.ts:143]: `fields` is a string array, but a `Locator` is appended here
```suggestion
        fields.push(paths[i]);
```

File: e2e-pw/src/oss/poms/field-visibility/field-visibility.ts [hunk 6]
```diff
@@ -63,4 +209,68 @@ class FieldVisibilityAsserter {
     private readonly fv: FieldVisibilityPom,
     private readonly sb: SidebarPom
```
Comment (benjaminpkane) [inline e2e-pw/src/oss/poms/field-visibility/field-visibility.ts:210]: nit: sb is unused
```suggestion
```

File: app/packages/components/src/components/TabOption/TabOption.tsx [hunk 7]
```diff
@@ -77,9 +78,9 @@ const TabOption = ({
       onMouseEnter={() => set({ background: theme.background.body })}
       onMouseLeave={() => set({ background: theme.background.level1 })}
     >
-      {options.map(({ text, title, onClick }, i) => (
+      {options.map(({ text, title, onClick, dataCy }, i) => (
         <Tab
-          data-cy={`tab-option-${title}`}
+          data-cy={dataCy}
```
Comment (benjaminpkane) [inline app/packages/components/src/components/TabOption/TabOption.tsx:83]: I think this change is breaking existing tests. Easy fix is make `tab-option-${title}` the fallback if `dataCy` is not provided
```suggestion
          data-cy={dataCy || `tab-option-${title}`}
```

File: e2e-pw/src/oss/poms/field-visibility/field-visibility.ts [hunk 8]
```diff
@@ -63,4 +209,68 @@ class FieldVisibilityAsserter {
...
   ) {}
+
+  async fieldVisibilityIconHasTooltip() {
```
Comment (benjaminpkane) [inline e2e-pw/src/oss/poms/field-visibility/field-visibility.ts:213]: Awesome assertions! 🔥 👏

File: e2e-pw/src/oss/poms/field-visibility/field-visibility.ts [hunk 9]
```diff
@@ -63,4 +209,68 @@ class FieldVisibilityAsserter {
...
+    await this.fv.fieldVisibilityBtn.hover();
+    await expect(this.fv.fieldVisibilityToggleTooltip).toBeVisible();
+  }
+
+  async assertAllFieldsSelected() {
```
Comment (benjaminpkane) [inline e2e-pw/src/oss/poms/field-visibility/field-visibility.ts:218]: A generic approach could have the `selectionFields` as an argument
```suggestion
  async assertAllFieldsSelected(selectionFields: string[]) {
```

**Review by benjaminpkane — CHANGES_REQUESTED** [review]
Spec and POM look awesome! Happy to see this merged once tests are passing in the workflow

File: e2e-pw/src/oss/poms/field-visibility/field-visibility.ts (hunk 9 above)
Comment (manivoxel51) [reply e2e-pw/src/oss/poms/field-visibility/field-visibility.ts:218]: good idea!

**Review by benjaminpkane — APPROVED** [review]
LGTM 🚀 I rebased onto `develop` to make sure e2e passes now that #3533 is merged

**Pull request merged**